### PR TITLE
Fix 573 597 600

### LINKED ISSUE #573: getMethod(String, CtTypeReference...) doesn't work with generics
With this snippet:

```java
public <T> void method1(T t) {
}

public <T extends String> void method1(T t) {
}
```

If I call `getMethod("method1", int.class)` on a `CtType`, it returns the wrong method.

### PR DESCRIPTION
**Pull request by GerardPaligot**
Fix #573 #597 #600

### CODE REVIEW

File: src/test/java/spoon/test/main/MainTest.java [hunk 1]
```diff
@@ -91,6 +92,21 @@ public void checkGenericContracts(CtPackage pack) {
...
+		// type parameter reference.
+		checkBoundAndUnboundTypeReference(pack);
+	}
+
+	private void checkBoundAndUnboundTypeReference(CtPackage pack) {
+		new CtScanner() {
+			@Override
+			public void visitCtTypeParameterReference(CtTypeParameterReference ref) {
+				if (!(ref.getParent() instanceof CtReference)) {
+					assertEquals(ref.getSimpleName(), ref.getDeclaration().getSimpleName());
```
Comment (monperrus) [inline src/test/java/spoon/test/main/MainTest.java]: only in Spoon mode:  (ref.getDeclaration!=null)

**Comment by monperrus** [comment]
CtTypeInformation {

   /** returns true if this element is a generics (eg "T") and false if it is an actual type (eg 'Book" or "String") */
   isGenerics() {
     ....
  }

}

**Comment by GerardPaligot** [comment]
@monperrus PR ok.

**Comment by monperrus** [comment]
could you add somewhere in a test a specification of isGenerics assertTrue(isGenerics) and assertFalse(isGenerics)?

**Pull request merged**